src/rabbit_delayed_message.erl

```diff
@@ -168,22 +168,28 @@ maybe_delay_first() ->
 
 route(#delay_key{exchange = Ex}, Deliveries, State) ->
     ExName = Ex#exchange.name,
-    lists:map(fun (#delay_entry{delivery = D}) ->
-                      D2 = swap_delay_header(D),
-                      Dests = rabbit_exchange:route(Ex, D2),
+    lists:map(fun (#delay_entry{delivery = Msg0}) ->
+                      Msg1 = case Msg0 of
+                               #delivery{message = BasicMessage} ->
+                                   mc_amqpl:from_basic_message(BasicMessage);
```

Comment:
You can probably just return `BasicMessage` as it is as `#basic_message{}` is a subtype of `mc:state()`